app/src/main/java/sdumchykov/androidApp/presentation/viewPager/contacts/MyContactsFragment.kt

```diff
@@ -0,0 +1,258 @@
+import sdumchykov.androidApp.presentation.viewPager.contacts.adapter.listener.UsersListener
+import sdumchykov.androidApp.presentation.viewPager.contacts.fetchContacts.FetchContacts
+
+private const val SNACKBAR_TIME_LENGTH = 5000
+
+@AndroidEntryPoint
+class MyContactsFragment :
+    BaseFragment<FragmentMyContactsBinding>(FragmentMyContactsBinding::inflate) {
+
+    private val pagerFragment by lazy { parentFragment as ViewPagerFragment }
+    private val parentViewModel by lazy { pagerFragment.myContactsViewModel }
+    private val usersAdapter: UsersAdapter by lazy {
+        UsersAdapter(
+            usersListener = object : UsersListener {
+                override fun onUserClickAction(userModel: UserModel, adapterPosition: Int) {
+                    Log.d("MainActivity", "onUserClickAction: ${userModel.id}")
+                    val action =
+                        ViewPagerFragmentDirections.actionViewPagerFragmentToContactProfileFragment(
+                            userModel
+                        )
+                    Navigation.findNavController(binding.root).navigate(action)
+                }
+
+                override fun onTrashIconClickAction(userModel: UserModel, userIndex: Int) {
+                    deleteContact(userModel, userIndex)
+                }
+
+                override fun onContactRemove(userModel: UserModel) {
+                    parentViewModel.removeItem(userModel)
+                }
+
+                override fun onMultiselectActivated() {
+                    usersAdapter.enableMultiselect()
+                    enableMultiselectViewState()
+                    parentViewModel.selectedEvent.value = true
+                    disableContactSwipe()
+                    initRecyclerView()
+                }
+
+                override fun onContactSelectedStateChanged() {
+                    if (usersAdapter.areAllItemsUnselected()) {
+                        disableMultiselectViewState()
+                        parentViewModel.selectedEvent.value = false
+                        enableContactSwipe()
+                        initRecyclerView()
+                    }
+                }
+            }
+        )
+    }
+
+    private var swipeFlags = ItemTouchHelper.END
+    private var dialogAddContact: AlertDialog? = null
+
+    //region Main code
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        initRecyclerView()
+        setSwipeToDelete()
+        createAddContactDialog()
+        handleRecyclerViewContent()
+    }
+
+    override fun setObservers() {
+        parentViewModel.userLiveData.observe(this) { users ->
+            usersAdapter.submitList(users.toMutableList())
+        }
+    }
+
+    override fun setListeners() {
+        imageButtonArrowBackSetOnClickListener()
+        multiSelectedItemsTrashCanOnClickListener()
```

Comment:
I suggest you can remove "SetOnClick" part, name will be shorter and still full-intented